### ISSUE TITLE
Godot master added input stuff into project.godot

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#19791]
Godot master ff01fd57bc19d9f4cb3929dbee87cdd9eda578e1

I noticed that testing Godot master on a 3.0.4 project did add a bunch of input stuff to project.godot, which were not there before... was it intented?

![image](https://user-images.githubusercontent.com/1311555/41935549-ff626d82-7981-11e8-8b18-b7f64239dd57.png)

**Comment by groud** [comment]
Yes. Inputs are not handled the same on master. There's a blog post about that.

Also, 3.1 is not backward compatible unless if you delete all those stuff for input.

**Comment by Zylann** [comment]
@groud I know they are different, but why are they suddenly getting saved into the project? That wasnt the case before it seems.

**Comment by groud** [comment]
Well, they were already saved but in another format, but only if you modified them. 

Since data was transformed between 3.0.4 and 3.1, I get they got saved anyway. I think we can consider this as normal yes.

**Comment by akien-mga** [comment]
If they're not modified, it might be better not to save them, as was the case in 3.0.x, no?

**Comment by groud** [comment]
Well, if you create a new project I don't think they will be added. It's only the conversion that did that.
The problem is that it's going to need a bunch of lines of code (comparing what's in the file to the default values) for something that is not really a problem.

**Comment by Zylann** [comment]
@groud sounds weird because my file didnt contain anything, in fact. So the only thing Godot would have seen was the defaults, but it still decided to save specifically those settings for some reason.

**Comment by groud** [comment]
Ok I just ran some test. In fact, if you change even a single mapping in the project settings, all mappings are saved (but this is not something I changed myself, it was already there).

Since the mappings get modified in the conversion process, they get saved too.
So either we implement a check to see if something is not the default value, and save it otherwise. Or we leave it as is, as it is not really important. ^^
